### DISCUSSION TITLE
Dual use bind? (Mac/Windows)

### DISCUSSION THREAD

**Discussion by 0x4A756E65** [nickcoutsos/keymap-editor#269] [Q&A] [answered]
Hi all, and thanks to @nickcoutsos for creating this!

I'm new here, so please bear with me. I use my Adv360 Pro with both Windows and Mac, and I have a very basic setup (see below) with minimal changes. My main modification is mapping ctrl to delete on the left thumb cluster.

However, I need this ctrl button to act as command on Mac. Is there a way to configure it so that, when connected to my Mac profile, it uses command instead of ctrl? Possibly some other way?

Thank you in advance!

![image](https://github.com/user-attachments/assets/a81587de-5f88-4597-b116-ce251934d4ff)

**Comment by nickcoutsos** [comment, marked as answer]
There isn't really a straightforward/automatic way. The root problem here is that USB devices (and Bluetooth, I think) have a reliable way to determine the OS of the host device.

If you want to tackle this purely on the keyboard, there's two parts:
1. Pick one OS mapping to be your "default" and create a layer for the other that remaps the key
2. Define a macro that activates your other profile and the additional layer, and another macro that switches back to the first profile and deactivates it.

This gets a little hairy, though, because there isn't reliable state management. There isn't a way to explicitly deactivate a particular layer. If you switch to a profile and activate a layer, and then later restart your keyboard you'll still have that profile active but the layer won't be. 

Another option is to do this remapping on the computer's side. You mentioned using macos, I know you can use Karabiner Elements and define a mapping that swaps Ctrl/Cmd (note that this would be for the keycode received by the computer and wouldn't know anything about whether it's the specific key assigned on your thumb cluster or anything). The benefit here is that you can limit this to your device so that if you're ever using a built-in or other external keyboard it'll do what's expected.

**Reply by 0x4A756E65** [reply]
Thank you so much @nickcoutsos - I am going to try some of this out and see what feels the best. I really appreciate your guidance!
